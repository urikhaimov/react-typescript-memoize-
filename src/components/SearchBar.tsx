import React from 'react';
import 'reactjs-popup/dist/index.css';


export interface Props {
    changeHandler(e: any): void;
}

const SearchBar: React.FC<Props> = ({ changeHandler }) => {
   

    return (

        <div className="search-bar">
           <label>Search
                <input id="searchbartext" onChange={changeHandler} />
            </label>
        </div>
    );
};

export default SearchBar;
