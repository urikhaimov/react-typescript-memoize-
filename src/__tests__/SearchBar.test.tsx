import React from 'react';
import renderer from 'react-test-renderer';
import SearchBar from '../components/SearchBar';


test('render App', () => {
    const filterCountries = jest.fn()
  const component = renderer.create(
    <SearchBar  changeHandler={filterCountries}/>,

  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

 
});