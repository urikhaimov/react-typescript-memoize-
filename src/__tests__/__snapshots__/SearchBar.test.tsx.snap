// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render App 1`] = `
<div
  className="search-bar"
>
  <form>
    <input
      id="searchbartext"
      onChange={[MockFunction]}
    />
  </form>
</div>
`;

exports[`render search Bar`] = `
<div
      className="search-bar"
    >
      <form>
        <input
          id="searchbartext"
         onChange={[MockFunction]}
        />
      </form>
    </div>
`;
